internal/pkg/aws/codepipeline/codepipeline.go

```diff
@@ -210,6 +213,43 @@ func (c *CodePipeline) ListPipelineNamesByTags(tags map[string]string) ([]string
 	return pipelineNames, nil
 }
 
+// pipelineExecutionID returns the ExecutionID of the most recent execution of a pipeline.
+func (c *CodePipeline) pipelineExecutionID(pipelineName string) (string, error) {
```

Comment:
nit: can we move the private methods below all the exported methods in the file?